datacontract/imports/dbt_importer.py

```diff
@@ -75,16 +75,33 @@ def import_dbt_manifest(
         if dbt_nodes and model_contents.name not in dbt_nodes:
             continue
 
+        model_unique_id = model_contents.unique_id
+
+        if isinstance(model_contents, ModelNode):
+            test_nodes = []
+            for node_id in manifest.child_map.get(model_unique_id, []):
+                test_node = manifest.nodes.get(node_id)
+                if not test_node or test_node.resource_type != "test":
+                    continue
+                if not isinstance(test_node, GenericTestNode):
+                    continue
+                if test_node.config.where is not None:
+                    continue
+                test_nodes.append(test_node)
+            primary_key = model_contents.infer_primary_key(test_nodes)
```

Comment:
Unfortunately, the `infer_primary_key` function only exists in the `ModelNode` class, so we have narrowed down the target to the `ModelNode` class only.